src/Hangfire.PostgreSql/PostgreSqlWriteOnlyTransaction.cs

```diff
@@ -55,17 +54,40 @@ public class PostgreSqlWriteOnlyTransaction : JobStorageTransaction
 
         public override void Commit()
         {
-            using (var transaction = _connection.BeginTransaction(IsolationLevel.RepeatableRead))
+            var isolationLevel = IsolationLevel.RepeatableRead;
+            var scopeOption = TransactionScopeOption.RequiresNew;
+            if (_options.EnableTransactionScopeEnlistment)
             {
+                var currentTransaction = Transaction.Current;
+                if (currentTransaction != null)
+                {
+                    isolationLevel = currentTransaction.IsolationLevel;
+                    scopeOption = TransactionScopeOption.Required;
+                }
+            }
+
+            var transactionOptions = new TransactionOptions()
+            {
+                IsolationLevel = isolationLevel,
+                Timeout = _options.TransactionSynchronisationTimeout
```

Comment:
Is there any chance that `scopeOption` was forgot to be set there?

Reply:
`scopeOption` ist set to `TransactionScopeOption.RequiresNew` in the top, so its the default in this method.
`RequiresNew` does not enlist in existing ambient transactions. This has also been verified via a unit test.